articlequality/feature_lists/ptwiki.py

```diff
@@ -0,0 +1,160 @@
+"""
+Portuguese Wikipedia
+++++++++++++++++++++
+"""
+import re
+
+from revscoring import Feature
+from revscoring.datasources.meta import filters, mappers
+from revscoring.features import wikitext
+from revscoring.features.meta import aggregators
+from revscoring.features.modifiers import log, max, sub
+from revscoring.languages import portuguese
+
+from . import wikipedia
+
+# Templates
+infobox_templates = wikitext.revision.template_names_matching(
+    r"Info", name="ptwiki.revision.infobox_templates")
+CN_TEMPLATES = [
+    r"Carece de fontes",
+    r"Carece de fontes2",
+    r"Carece de fontes/bloco",
+    r"Carece de fontes/bloco2"
+]
+cn_templates = wikitext.revision.template_names_matching(
+    "|".join(CN_TEMPLATES), name="ptwiki.revision.cn_templates")
+main_article_templates = wikitext.revision.template_names_matching(
+    "Main", name="ptwiki.main_article_templates")
+CITE_TEMPLATES = [
+    r"Cite",
+    r"Harvard[_ ]citation[_ ]no[_ ]brackets", r"harvnb",
+    r"Harvard citation", r"harv",
+    r"Harvard citation text", r"harvtxt",
+    r"Harvcoltxt",
+    r"Harvcol",
+    r"Harvcolnb",
+    r"Harvard citations", r"harvs",
+    r"Harvp"
+]
+cite_templates = wikitext.revision.template_names_matching(
+    "|".join(CITE_TEMPLATES), name="ptwiki.revision.cite_templates")
+SFN_TEMPLATES = [
+    r"Shortened footnote template", r"sfn",
+    r"Sfnp",
+    r"Sfnm",
+    r"Sfnmp"
+]
+shortened_footnote_templates = wikitext.revision.template_names_matching(
+    "|".join(SFN_TEMPLATES),
+    name="ptwiki.revision.shortened_footnote_templates")
+all_ref_tags = shortened_footnote_templates + wikitext.revision.ref_tags
+all_cite_templates = cite_templates + shortened_footnote_templates
+proportion_of_templated_references = \
+    all_cite_templates / max(all_ref_tags, 1)
+non_templated_references = max(all_ref_tags - all_cite_templates, 0)
+non_cite_templates = sub(
+    wikitext.revision.templates, all_cite_templates,
+    name="ptwiki.revision.non_cite_templates"
+)
+
+# Links
+category_links = wikitext.revision.wikilink_titles_matching(
+    r"Category\:", name="ptwiki.revision.category_links")
+
+image_links = wikitext.revision.wikilink_titles_matching(
+    r"File|Image\:", name="ptwiki.revision.image_links")
+
+image_templates = wikitext.revision.template_names_matching(
+    r"((Wide|Tall|scalable) image)|Panorama|Panorama 2",
+    name='ptwiki.revision.image_template')
+
+
+def get_images(strs):
+    """
+    Parses a list of strings, expected to be tags or templates
+    to get matching image patterns and returns a count of the matches.
+    """
+    matches = re.findall(
+        r"image[1-9]{1,2}|File:|Image:|photo[1-9][a-z]",
+        "".join(strs), re.I)
+    return len(matches)
```

Comment:
Add "Ficheiro" as a translation of "File" and "Arquivo" as it is an alias.  "imagem" is the alias of "image".  

I'd change this to `r"imagem?[1-9]{1,2}|(File|Ficheiro|Image|Imagem|Archivo)\:"`